cmd/sidecars/network-passt-binding/server/server.go

```diff
@@ -88,7 +88,7 @@ func (s V1alpha3Server) OnDefineDomain(_ context.Context, params *hooksV1alpha3.
 		IstioProxyInjectionEnabled: istioProxyInjectionEnabled,
 	}
 
-	passtConfigurator, err := domain.NewPasstNetworkConfigurator(vmi.Spec.Domain.Devices.Interfaces, vmi.Spec.Networks, opts)
+	passtConfigurator, err := domain.NewPasstNetworkConfigurator(vmi.Spec.Domain.Devices.Interfaces, vmi.Spec.Networks, opts, nil)
```

Comment:
Shouldn't it have the concrete link discoverer?

Reply:
On nill it fallback to netlink, so everything is fine.

Reply:
Oh, I missed that part.

nit: I would pass the concrete link-discoverer here (&netlink.NetLink{}`), instead of having `NewPasstNetworkConfigurator` account for it.
IMO it would also make the code cleaner and more predictable.